ink_linting/ui/pass/mapping-two-constructors.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright 2018-2022 Parity Technologies (UK) Ltd.
+// This file is part of cargo-contract.
+//
+// cargo-contract is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// cargo-contract is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with cargo-contract.  If not, see <http://www.gnu.org/licenses/>.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use ink_lang as ink;
+
+#[ink::contract]
+mod my_contract {
+    use ink_storage::{traits::SpreadAllocate, Mapping};
+
+    #[ink(storage)]
+    #[derive(SpreadAllocate)]
+    pub struct MyContract {
+        balances: Mapping<AccountId, Balance>,
+    }
+
+    impl MyContract {
+        /// The `initialize_contract` must be found, even if there is logic before it.
```

Comment:
Is this a copy of the "additional logic" UI test? There is only a single constructor here where there should be two according to the test name.